docs/integration_fetching_data.md

```diff
@@ -0,0 +1,146 @@
+
+Home Assistant provides a Data Update Coordinator class to help you manage this as efficient as possible.
+
+```python
+import logging
+
+from homeassistant.helpers import debounce, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+
+_LOGGER = logging.getLogger(__name__)
+
+async def create_coordinator(api):
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await api.fetch_data()
+        except ApiError:
+            raise UpdateFailed
+
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        "sensor",
+        async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        timedelta(seconds=30),
+        # Debouncer to limit request_refresh
+        debounce.Debouncer(
+            hass,
+            _LOGGER,
+            # Timeout to wait between refreshes
+            0.3,
+            # When a refresh is requested, should we refresh right await and then wait
+            # Or should we wait and refresh at the end of timeout.
+            True
+        ),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    return coordinator
+
+
+class MyEntity(entity.Entity):
+
+    def __init__(self, coordinator):
+        self.coordinator = coordinator
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return not self.coordinator.failed_last_update
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.coordinator.async_add_listener(
+            self.async_write_ha_state
+        )
+
+    async def async_will_remove_from_hass(self):
+        """When entity will be removed from hass."""
+        self.coordinator.async_remove_listener(
+            self.async_write_ha_state
+        )
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on.
+
+        Example method how to request data updates.
+        """
+        # Do the turning on.
+        # ...
+
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_update(self):
+        """Update the entity.
+
+        Only used by the generic entity update service.
+        """
+        await self.coordinator.async_request_refresh()
+```
+
+## One API endpoint per device
+
+If your one API endpoint is mapping to multiple entities, please see the previous section. Even if there are many endpoints, it's best to use a coordinator per endpoint to manage this data.
```

Comment:
I think this raises more questions than it answers. Maybe we should simply have a warning here pointing towards using coordinator always, assuming that is the preferred way to go?